### ISSUE TITLE
Set service worker version

### ISSUE THREAD

**Issue by hrishikesh-k** [Voxylu/gatsby-plugin-sw#3]
How we can set the version of the service worker cache? Currently, because of cache, my website loads older versions of the content. Thus, I'd like to check for service worker version to load new content. This is my first Gatsby website, so, I am not sure how to add the version in the service worker and the contents of the sw.js are generated automatically during export.

My current sw.js contents:

```
importScripts(
    'https://storage.googleapis.com/workbox-cdn/releases/5.1.2/workbox-sw.js'
  )
  workbox.precaching.precacheAndRoute([])
```

P.S.: I haven't yet tested the build with the new (5.1.2) version. I just happened to find that verion now, my current live build uses the version mentioned in the docs of the plugin (3.4.1). I'll publish another build to the website, tomorrow, and will test the version update then, hopefully, shouldn't cause any problems.

**Comment by hrishikesh-k** [comment]
Okay, I read more about Workbox and found out that it adds a revision attribute to all files. So, all I had to do was, add this `exports.onServiceWorkerUpdateReady= () => window.location.reload(true);` to my `gatsby-browser.js`.

This fixes the content not refreshing problems, but, it's not deleting the previous cache, and thus, my service worker is now using more storage on the device. I'll research more on how to fix it.

**Comment by hrishikesh-k** [comment]
Final update: I think I've finished my research and got a robust working solution. I'll detail it all if someone else needs it.

**Problem 1: Service worker's outdated cache:**

My initial problem was that, because of the cache generated by service worker, the updated versions of my website didn't load. Users were seeing the cached versions. I found out that, if I can update the service worker's version, maybe, the new service worker will get installed and thus, add new cache and as a result, users can see the updated version of my website. That's when I opened this issue.

**Solution:** I found out that Workbox (the tool that this plugin uses to generate the service worker), handles the versioning automatically by adding a revision attribute to the assets. Thus, only the files that are updated have to get re-cached as the unchanged files have the same hash (revision attribute).

That made me think that, my website should show the latest version, but, that wasn't happening. So, I ran into the next problem.

**Problem 2: Deleting the old cache**

I needed to delete the old cache.

**Solution:** The simplest way to do it was to add `cleanupOutdatedCaches: true` to my `sw.js`.

This deleted the old cache thus making sure your users always have just one copy of the cache and thus, no extra storage space is used. This is just a preventive way to not end up with a lot of cache.

Lastly, I needed to find a way to install and activate the new service worker immediately. Thus, the next problem.

**Problem 3: Install and activate the new service worker immediately**

If the new service worker will install immediately, the cache will be filled by the latest versions in the least time.

**Solution:** Add `skipWaiting();` to the `sw.js`.

So, my final `sw.js` looks like this:

```
importScripts(
    'https://storage.googleapis.com/workbox-cdn/releases/5.1.2/workbox-sw.js'
  )
  workbox.precaching.precacheAndRoute([])
  cleanupOutdatedCaches: true
  skipWaiting();
```

Now, everything finally works as expected. After I publish my website, the first refresh in the browser loads the old content, installs the new service worker and caches the new files. On next refresh, the new content loads. The websie continues to load cached assets when available and also works offline.
